### PR TITLE
Cura-6683 icons per object settings

### PR DESCRIPTION
**Pull request by ninovanhooff**

### CODE REVIEW

File: plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml [hunk 1]
```diff
@@ -28,490 +73,478 @@ Item {
 
         spacing: UM.Theme.getSize("default_margin").height
 
-        Row
+        Component.onCompleted: setMeshType(UM.ActiveTool.properties.getValue("MeshType"))
```
[the remark this answers is not in the document]
Comment (ninovanhooff) [reply plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml]: This is an initialisation line. If I remove it, no button is selected and the mesh type label has no value.

[the remark this answers is not in the document]
Comment (ninovanhooff) [reply plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml]: I now used onCurrent_mesh_typeChanged to trigger a view update, which makes this line indeed spurious. I think this was your intention :-)

File: plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml [hunk 2]
```diff
@@ -28,123 +71,96 @@ Item {
...
-                        if(meshTypeSelection.model.get(index).type == mesh_type)
-                        {
-                            meshTypeSelection.currentIndex = index;
-                            return;
-                        }
-                    }
-                    meshTypeSelection.currentIndex = 0;
-                }
+                id: overhangMeshButton
+                text: catalog.i18nc("@label", "Modify settings for overlays")
```
Comment (diegopradogesto) [inline plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml]: Text should be "Modify settings for overlaps"
```suggestion
                text: catalog.i18nc("@label", "Modify settings for overlaps")
```

File: plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml [hunk 3]
```diff
@@ -4,22 +4,65 @@
...
+        {
+            setMeshType(cutting_mesh_type)
+        }
+    }
+
+    function setMeshType(type){
+        UM.ActiveTool.setProperty("MeshType", type)
+    }
+
+    function updateView(type) {
```
[the remark this answers is not in the document]
Comment (ninovanhooff) [reply plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml]: Not sure how, but I also think it is more readable this way.

File: plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml [hunk 4]
```diff
@@ -4,22 +4,65 @@
...
+    readonly property string normal_mesh_type: ""
+    readonly property string support_mesh_type: "support_mesh"
+    readonly property string cutting_mesh_type: "cutting_mesh"
+    readonly property string infill_mesh_type: "infill_mesh"
+    readonly property string anti_overhang_mesh_type: "anti_overhang_mesh"
+
+    property var current_mesh_type: UM.ActiveTool.properties.getValue("MeshType")
+
+
+    onCurrent_mesh_typeChanged: updateView(current_mesh_type)
```
[the remark this answers is not in the document]
Comment (ninovanhooff) [reply plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml]: Gladly!

File: plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml [hunk 5]
```diff
@@ -28,123 +71,96 @@ Item {
...
             {
-                text: catalog.i18nc("@label","Mesh Type")
-                font: UM.Theme.getFont("default")
-                color: UM.Theme.getColor("text")
-                height: UM.Theme.getSize("setting").height
-                verticalAlignment: Text.AlignVCenter
+                id: normalButton
+                text: catalog.i18nc("@label", "Normal model")
+                iconSource: UM.Theme.getIcon("pos_normal");
+                property bool needBorder: true
```
[the remark this answers is not in the document]
Comment (ninovanhooff) [reply plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml]: Discussed with Yi-An. We will not use the thick line anymore

File: plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml [hunk 6]
```diff
@@ -159,26 +175,26 @@ Item {
...
-                            return {"settable_per_mesh": true};
+                            return {"settable_per_mesh": true}
                         }
                         exclude:
                         {
-                            var excluded_settings = [ "support_mesh", "anti_overhang_mesh", "cutting_mesh", "infill_mesh" ];
+                            var excluded_settings = [ "support_mesh", "anti_overhang_mesh", "cutting_mesh", "infill_mesh" ]
 
-                            if(meshTypeSelection.model.get(meshTypeSelection.currentIndex).type == "support_mesh")
+                            if(current_mesh_type == "support_mesh")
```
[the remark this answers is not in the document]
Comment (Ghostkeeper) [reply plugins/PerObjectSettingsTool/PerObjectSettingsPanel.qml]: There is no code style rule about this. In general I'm preferring without space but either way is fine.

**Pull request merged**